src/main/java/org/protege/editor/owl/server/api/ServerLayer.java

```diff
@@ -61,10 +69,30 @@ public void createCodegenFile(String projectId) throws IOException {
         os.write("1000".getBytes());
     }
     
-    public String  getHistoryFilePath(Project proj) throws IOException {
-    	
-    	HistoryFile f = createHistoryFile(proj.getId().get(), proj.getName().get());
+    public String getHistoryFilePath(Project proj) throws IOException {
+    	HistoryFile f = createHistoryFile(proj.getId().get());
     	
     	return f.getAbsolutePath();
     }
+
+    public void saveProjectSnapshot(SnapShot snapshot, File snapshotFile) throws IOException {
+        BufferedOutputStream outputStream = new BufferedOutputStream(new FileOutputStream(snapshotFile));
+        try {
+            BinaryOWLOntologyDocumentSerializer serializer = new BinaryOWLOntologyDocumentSerializer();
+            long start = System.currentTimeMillis();
+            serializer.write(new OWLOntologyWrapper(snapshot.getOntology()), new DataOutputStream(outputStream));
+            logger.info("Saving snapshot in " + (System.currentTimeMillis() - start) + " ms");
+        }
+        finally {
+            if (outputStream != null) {
```

Comment:
tWhy do you have this null check here? Seems like it will never be a null unless you assign a null explicitly to it